### ISSUE TITLE
Voting has stopped working

### ISSUE THREAD

**Issue by e40** [dangwu/Octal#295]
Within the last few days I started getting "Something went wrong, Please try again later" and the vote was reverted (either up or down).  iOS 17.2.1.  Octal 4.15.1.

![IMG_3416](https://github.com/dangwu/Octal/assets/89416/65e2ad15-96a9-4a52-89b9-b13c22ea9be3)

**Comment by dangwu** [comment]
I just tested and it's working for me. Can you try upvoting on the web and
seeing if you have some account issue?

**Comment by e40** [comment]
> I just tested and it's working for me. Can you try upvoting on the web and seeing if you have some account issue?

I upvote on the web all the time.  No issue.

**Comment by e40** [comment]
![Screenshot 2024-01-10 at 9 28 24 AM](https://github.com/dangwu/Octal/assets/89416/bb4c4707-051d-418d-a9c3-c4845c3c77c0)

**Comment by gingerbeardman** [comment]
Up vote works for me. 

Down vote gives me an error.

**Comment by dangwu** [comment]
Downvoting works for me. Unfortunately this is another issue that I can't really debug - especially if I can't repro it 😭

**Comment by gingerbeardman** [comment]
It seems related to voting after a bunch of other actions (topic load or refresh, another vote, etc).

So some sort of rate limiting?

Does the API not return any more helpful error message or info than "something went wrong"?

Repro
- Just keep hammering the up/down vote button to toggle the vote. Quite quickly you'll get the error.

**Comment by dangwu** [comment]
Yes if you perform write actions quickly you'll get rate limited. Voting is performed by sending HN HTTP requests - not using the official API. I used to try to parse the HTML responses to look for error messages, but that's incredibly brittle and broke constantly with HTML changes. Rate limiting errors are still "successful" page loads thus there's no status code to look for either.

**Comment by gingerbeardman** [comment]
Understood, thanks for your continued work on this app — it rocks regardless!
